### PR TITLE
Record Labels and Env from pushed docker images

### PR DESCRIPTION
**Pull request by ashb**
This will le us know what version of Airflow that a docker image claims to be. (Claim, as this could be faked by end-user docker image if they so desired).

Right now we don't do anything with this information. I have tested this on a locally running docker registry without auth, so I'm not 100% sure I've got the auth/JWT right for the out-bound requests

### CODE REVIEW

**Comment by ashb** [comment]
Here's what we store. Lables are k/v pairs, env is just a list of strings. It _may_ make sense to turn env in to a k/v pairs before we store it too.

```
prisma=# select id,name, tag,digest,"createdAt",deployment, jsonb_pretty(labels::jsonb),jsonb_pretty(env::jsonb) from houston$default."DockerImage";
-[ RECORD 1 ]+------------------------------------------------------------------------------------------------------------------------
id           | ck29gas0d00ld0760ltjidx7x
name         | telescopic-sun-2787/airflow:cli-1
tag          | cli-1
digest       | sha256:907b4a633d31872f7fc9b4cb22998b9de4c25f8cc8f08529ca56c2ace698e541
createdAt    | 2019-10-27 20:33:38.744
deployment   | ck297jx9y00i807603weklx6e
jsonb_pretty | {                                                                                                                      +
             |     "maintainer": "Astronomer <humans@astronomer.io>",                                                                 +
             |     "io.astronomer.docker": "true",                                                                                    +
             |     "io.astronomer.docker.module": "airflow",                                                                          +
             |     "io.astronomer.docker.component": "airflow",                                                                       +
             |     "io.astronomer.docker.build.number": "1403",                                                                       +
             |     "io.astronomer.docker.airflow.version": "1.10.5"                                                                   +
             | }
jsonb_pretty | [                                                                                                                      +
             |     "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",                                               +
             |     "AIRFLOW_REPOSITORY=https://github.com/astronomer/airflow",                                                        +
             |     "AIRFLOW_MODULE=git+https://github.com/astronomer/airflow@1.10.5-1#egg=apache-airflow[all, statsd, elasticsearch]",+
             |     "AIRFLOW_HOME=/usr/local/airflow",                                                                                 +
             |     "PYMSSQL_BUILD_WITH_BUNDLED_FREETDS=1",                                                                            +
             |     "PYTHONPATH=/usr/local/airflow",                                                                                   +
             |     "AIRFLOW_GPL_UNIDECODE=True",                                                                                      +
             |     "ASTRONOMER_USER=astro",                                                                                           +
             |     "ASTRONOMER_GROUP=astro",                                                                                          +
             |     "GUNICORN_CMD_ARGS=--preload"                                                                                      +
             | ]
```

**Comment by ashb** [comment]
Found a way of testing this locally. Auth not right yet.

File: database/datamodel.graphql [hunk 1]
```diff
@@ -126,4 +126,18 @@ type Deployment {
...
+  id: ID! @id
+  # Even though we only "need" the tag, to make it a unique string that we can
+  # create/update we include repository in this too.
+  name: String @unique
+  labels: Json!
+  env: Json!
+  tag: String!
+  digest: String!
+  deployment: Deployment! @relation(name: "DeploymentDockerImages" onDelete: CASCADE)
+  createdAt: DateTime! @createdAt
```
Comment (andriisoldatenko) [inline database/datamodel.graphql]: may be airflow version?

Comment (andriisoldatenko) [reply database/datamodel.graphql]: `airflowVersion`?

**Comment by schnie** [comment]
@ashb think we're good to merge this and build upon it. Looks like we got a few small conflicts though.

**Comment by ashb** [comment]
Conflicted with my own PR :D  Fixing now.

**Comment by ashb** [comment]
@schnie Conflicts resolved.

File: database/datamodel.graphql (hunk 1 above)
Comment (ashb) [reply database/datamodel.graphql]: Decided not for now, can (will) add this later once we've worked out the UI around it.

**Pull request merged**